src/controllers/expenses-Controllers.js

```diff
@@ -0,0 +1,81 @@
+const expenseService = require('../services/expenses-Services');
+
+const getAllExpenses = async (req, res) => {
+  const { userId, from, to, categories: category } = req.query;
+
+  const filteredExpenses = await expenseService.getAllExpenses({
+    userId,
+    from,
+    to,
+    category,
+  });
+
+  res.send(filteredExpenses);
+};
+
+const getExpensById = async (req, res) => {
+  const { id } = req.params;
+
+  const normalizedId = +id;
+  const targetExpense = await expenseService.getExpenseById(normalizedId);
+
+  if (targetExpense.error) {
+    return res.sendStatus(404);
+  }
+
+  return res.send(targetExpense.data);
+};
+
+const createExpense = async (req, res) => {
+  const { userId, spentAt, title, amount, category, note } = req.body;
+
+  const result = await expenseService.createExpense(
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  );
+
+  if (result.error) {
+    return res.sendStatus(400);
+  }
+
+  return res.status(201).send(result.data);
+};
```

Comment:
You should check on controller level if you have all required arguments in req.body and if not, just send 400 status.

Reply:
The same for all controllers you use in the app